visualization/app/codeCharta/ui/copyToClipboardButton/util/getFilenamesWithHighestMetric.spec.ts

```diff
@@ -0,0 +1,137 @@
+import { NodeType } from "../../../codeCharta.model"
+import { FileToValue, getFilenamesWithHighestMetrics, updateAttributeMap } from "./getFilenamesWithHighestMetrics"
+
+let MAP: Map<string, FileToValue[]>
+
+beforeEach(() => {
+	MAP = new Map<string, FileToValue[]>([
+		[
+			"mcc",
+			[
+				{ name: "file2", value: 500 },
+				{ name: "file3", value: 400 },
+				{ name: "file1", value: 300 },
+				{ name: "file1", value: 200 },
+				{ name: "file1", value: 100 },
+				{ name: "file1", value: 50 },
+				{ name: "file1", value: 20 },
+				{ name: "file1", value: 15 },
+				{ name: "file1", value: 10 },
+				{ name: "file1", value: 5 }
+			]
+		]
+	])
+})
+
+describe("getFilenamesWithHighestMetrics", () => {
+	it("should give map which has exactly two keys (mcc, rloc)", () => {
+		const resultMap = getFilenamesWithHighestMetrics(BIG_NODE_WITH_TWO_ATTRIBUTES)
+		const mccResults = resultMap.get("mcc")
+		const rlocResults = resultMap.get("rloc")
+
+		expect(resultMap.size).toBe(2)
+		expect(mccResults).toBeTruthy()
+		expect(rlocResults).toBeTruthy()
+	})
+	it("should restrict to at most 10 files for each attribute", () => {
+		const resultMap = getFilenamesWithHighestMetrics(BIG_NODE_WITH_TWO_ATTRIBUTES)
+
+		for (const files of resultMap) {
+			expect(files.length).toBeLessThanOrEqual(10)
+		}
+	})
+	it("should return correct values for rloc and mcc", () => {
+		const resultMap = getFilenamesWithHighestMetrics(BIG_NODE_WITH_TWO_ATTRIBUTES)
+		const mccResults = resultMap.get("mcc")
+		const rlocResults = resultMap.get("rloc")
+
+		expect(rlocResults).toEqual([{ name: "leaf1", value: 5 }])
+		expect(mccResults).toEqual([
+			{ name: "leaf1", value: 10 },
+			{ name: "leaf2", value: 9 },
+			{ name: "leaf3", value: 8 },
+			{ name: "leaf4", value: 7 },
+			{ name: "leaf5", value: 6 },
+			{ name: "leaf6", value: 5 },
+			{ name: "leaf7", value: 4 },
+			{ name: "leaf8", value: 3 },
+			{ name: "leaf9", value: 2 },
+			{ name: "leaf10", value: 1 }
+		])
+	})
+	it("should ignore folders", () => {
+		const map = getFilenamesWithHighestMetrics(CONTAINS_FOLDER_WITH_ATTRIBUTES)
+
+		expect(map.get("rloc")).toBeUndefined()
+		expect(map.get("mcc")).toEqual([{ name: "file", value: 0 }])
+	})
+})
+
+describe("updateAttributeMap", () => {
+	it("should not add if value is too low", () => {
```

Comment:
```suggestion
	it("should not add if value not one of the 10 highest", () => {
```

Reply:
"not one of the 10 highest" might be outdated soon. How about "is the lowest value"?